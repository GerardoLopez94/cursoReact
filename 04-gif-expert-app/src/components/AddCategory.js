import React, {useState} from 'react';

const AddCategory = () =>{

    const [inputValue, setInutValue] = useState('Hola Mundo');

    const handleInputChange = (e) =>{
        setInutValue(e.target.value)
    }

    const handleSubmit = (e) =>{
        //Prevenimos que el navegador se refresque
        e.preventDefault();

        console.log('Submit Hecho')
    }

    return (
        <form onSubmit={handleSubmit}>
            <input 
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

export { AddCategory };