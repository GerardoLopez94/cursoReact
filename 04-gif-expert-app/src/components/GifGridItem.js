import React from 'react';

const GifGridItem = ({img}) =>{
    return (
        <div>
            { img.title }
        </div>
    )

}

export { GifGridItem };