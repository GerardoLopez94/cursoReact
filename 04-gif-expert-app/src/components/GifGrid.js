import React from 'react';

import { useFetchGifs } from '../hooks/useFetchGifs';

import { GifGridItem } from './GifGridItem';






const GifGrid = ({ category }) =>{

    //const [images, setImages] = useState([]);
    const {loading, data: images} = useFetchGifs(category);

    return (
        <>
        <h3 className='animate__bounceIn'> { category } </h3>
        <br />
        { loading && <p>Loading</p> }
        <div className='card-grid'>
           
            {
                images.map( (img) => 
                <GifGridItem 
                key={img.id} 
                {...img}
                />
                )
            }
           
        </div>
        </>
    )
}

export { GifGrid };