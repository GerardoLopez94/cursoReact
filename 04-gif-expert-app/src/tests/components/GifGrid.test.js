import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('pruebas en <GifGrid />', () =>{
   

    test('debe de mostrar correctamente', () =>{

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category='Goku' />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://google.com',
            title: 'ABC Title'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category='Goku' />);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').exists()).toBe(true);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });


});