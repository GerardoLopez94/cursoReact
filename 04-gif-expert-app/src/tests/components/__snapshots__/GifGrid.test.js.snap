// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGrid /> debe de mostrar correctamente 1`] = `
<Fragment>
  <h3
    className="animate__bounceIn"
  >
     
    Goku
     
  </h3>
  <br />
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`pruebas en <GifGrid /> debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__bounceIn"
  >
     
    Goku
     
  </h3>
  <br />
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="ABC Title"
      url="https://google.com"
    />
  </div>
</Fragment>
`;
